src/Sylius/Bundle/ApiBundle/StateProvider/Admin/ResetPasswordProvider.php

```diff
@@ -0,0 +1,35 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Sylius Sp. z o.o.
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Sylius\Bundle\ApiBundle\StateProvider\Admin;
+
+use ApiPlatform\Metadata\Operation;
+use ApiPlatform\State\ProviderInterface;
+use Sylius\Bundle\CoreBundle\Message\Admin\Account\ResetPassword;
+use Webmozart\Assert\Assert;
+
+/** @experimental */
+final class ResetPasswordProvider implements ProviderInterface
+{
+    public function provide(Operation $operation, array $uriVariables = [], array $context = []): ResetPassword
+    {
+        Assert::true($this->supportsClass($operation->getClass()));
```

Comment:
```suggestion
        Assert::same($operation->getClass(), ResetPassword::class);
```
I think this should be enough

Reply:
$operation returns string, that's why i use is_a() method

Reply:
I'm more convinced to is_a() when checking classes